invokeai/frontend/src/features/parameters/store/generationSlice.ts

```diff
@@ -71,12 +73,38 @@ const initialGenerationState: GenerationState = {
 
 const initialState: GenerationState = initialGenerationState;
 
+const loraExists = (state: GenerationState, lora: string) => {
+  const lora_regex = new RegExp(`withLora\\(${lora},?\\s*([^\\)]+)?\\)`);
+  if (state.prompt.match(lora_regex)) return true;
+  return false;
+};
+
 export const generationSlice = createSlice({
   name: 'generation',
   initialState,
   reducers: {
     setPrompt: (state, action: PayloadAction<string | InvokeAI.Prompt>) => {
       const newPrompt = action.payload;
+
+      // Tackle User Typed Lora Syntax
+      let lorasInUse = [...state.lorasInUse]; // Get Loras In Prompt
```

Comment:
This should really be split out of `setPrompt` and debounced. It's not obvious, but this will cause input lag for prompts when there is a large number of LoRA.

Reply:
I agree. I'm a tad bit busy but I'll to get to it. Maybe @psychedelicious can have a look if he's free too.